integration-libs/opf/base/opf-api/adapters/opf-api-base.adapter.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import {
+  ConverterService,
+  LoggerService,
+  backOff,
+  isJaloError,
+  normalizeHttpError,
+} from '@spartacus/core';
+import {
+  OPF_ACTIVE_CONFIGURATION_NORMALIZER,
+  OpfBaseAdapter,
+  OpfEndpointsService,
+} from '@spartacus/opf/base/core';
+import {
+  ActiveConfiguration,
+  OPF_CC_PUBLIC_KEY,
+  OpfConfig,
+} from '@spartacus/opf/base/root';
+import { Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable()
+export class OpfApiBaseAdapter implements OpfBaseAdapter {
+  protected logger = inject(LoggerService);
+
+  constructor(
+    protected http: HttpClient,
+    protected converter: ConverterService,
+    protected opfEndpointsService: OpfEndpointsService,
+    protected config: OpfConfig
+  ) {}
+
+  protected headerWithNoLanguage: { [name: string]: string } = {
+    accept: 'application/json',
+    'Content-Type': 'application/json',
+  };
+  protected header: { [name: string]: string } = {
+    ...this.headerWithNoLanguage,
+    'Accept-Language': 'en-us',
+  };
+
+  protected headerWithContentLanguage: { [name: string]: string } = {
+    ...this.headerWithNoLanguage,
+    'Content-Language': 'en-us',
+  };
+
+  getActiveConfigurations(): Observable<ActiveConfiguration[]> {
+    const headers = new HttpHeaders(this.header).set(
+      OPF_CC_PUBLIC_KEY,
+      this.config.opf?.commerceCloudPublicKey || ''
+    );
+
+    return this.http
+      .get<ActiveConfiguration[]>(this.getActiveConfigurationsEndpoint(), {
+        headers,
+      })
+      .pipe(
+        catchError((error) => {
+          throw normalizeHttpError(error, this.logger);
+        }),
+        backOff({
+          shouldRetry: isJaloError,
+        }),
+        this.converter.pipeable(OPF_ACTIVE_CONFIGURATION_NORMALIZER)
```

Comment:
[MEDIUM]

If you want to use a singular-sounding normalizer (i.e. `CONFIGURATION` instead of `CONFIGURATIONS`) on an `Array` response, please use the method `pipeableMany()`.

Reply:
Fixed.